### PR TITLE
[PATCH API-NEXT v3] AES-CTR support

### PR DESCRIPTION
**Pull request by lumag**

### CODE REVIEW

**Comment by lumag** [comment]
@Bill-Fischofer-Linaro Actually no. I implement tests for optional feature, making sure that they do not break the build if implementation provides no support for this feature, then I implement the feature.

AES-CTR provides an example of IPsec cipher, where length(salt) + length(IPSEC_IV) != length(cipher_IV).

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag OK, thanks. Still needs @psavol review for API change, but I'm Ok with this.

File: include/odp/api/spec/crypto.h [hunk 1]
```diff
@@ -90,7 +90,10 @@ typedef enum {
 	ODP_DEPRECATE(ODP_CIPHER_ALG_AES128_CBC),
 
 	/** @deprecated  Use ODP_CIPHER_ALG_AES_GCM instead */
-	ODP_DEPRECATE(ODP_CIPHER_ALG_AES128_GCM)
+	ODP_DEPRECATE(ODP_CIPHER_ALG_AES128_GCM),
+
+	/** AES with counter mode */
+	ODP_CIPHER_ALG_AES_CTR,
```
Comment (psavol) [inline include/odp/api/spec/crypto.h]: I'd move this after AES_CBC or AES_GCM. It's easier to see what's deprecated when those are on the bottom.

File: include/odp/api/spec/crypto.h [hunk 2]
```diff
@@ -169,6 +172,9 @@ typedef union odp_crypto_cipher_algos_t {
 		/** @deprecated  Use aes_gcm instead */
 		uint32_t ODP_DEPRECATE(aes128_gcm) : 1;
 
+		/** ODP_CIPHER_ALG_AES_CTR */
+		uint32_t aes_ctr     : 1;
+
```
Comment (psavol) [inline include/odp/api/spec/crypto.h]: Same here, move upwards. On top of deprecated.

File: include/odp/api/spec/crypto.h (hunk 1 above)
Comment (lumag) [reply include/odp/api/spec/crypto.h]: Doing so would break binary compatibility. Is it ok?

Comment (psavol) [reply include/odp/api/spec/crypto.h]: Yes, since ABI compatibility is promised to be maintained within an API version, but not between API versions.

**Comment by lumag** [comment]
@psavol updated to follow your feedback.

**Comment by lumag** [comment]
@muvarov could you please merge it, I'd like to submit depending patches.

**Comment by muvarov** [comment]
Merged.

**Pull request closed without merging**